tasks/release.py

```diff
@@ -35,6 +35,127 @@ def add_prelude(ctx, version):
+
+    agent_version_int = list(map(int, agent_version.split(".")))
+
+    if len(agent_version_int) != 3:
+        print("Error: invalid version: {}".format(agent_version_int))
+        raise Exit(1)
+
+    # let's avoid losing uncommitted change with 'git reset --hard'
+    try:
+        ctx.run("git diff --exit-code HEAD", hide="both")
+    except Failure:
+        print("Error: You have uncommitted changes, please commit or stash before using update-dca-changelog")
+        return
+
+    # make sure we are up to date
+    ctx.run("git fetch")
+
+    # let's check that the tag for the new version is present (needed by reno)
+    try:
+        ctx.run("git tag --list | grep dca-{}".format(new_version))
+    except Failure:
+        print("Missing 'dca-{}' git tag: mandatory to use 'reno'".format(new_version))
+        raise
+
+    # Cluster agent minor releases are in sync with the agent's, bugfixes are not necessarily.
+    # We rely on the agent's devel tag to enforce the sync between both releases.
+    branching_point_agent = "{}.{}.0-devel".format(agent_version_int[0], agent_version_int[1])
+    previous_minor_branchoff = "dca-{}.{}.X".format(new_version_int[0], new_version_int[1] - 1)
+    log_result = ctx.run(
+        "git log {}...remotes/origin/{} --name-only --oneline | \
+            grep releasenotes-dca/notes/ || true".format(
+            branching_point_agent, previous_minor_branchoff
+        )
+    )
+    log_result = log_result.stdout.replace('\n', ' ').strip()
+
+    # Do not include release notes that were added in the previous minor release branch (previous_minor_branchoff)
+    # and the branch-off points for the current release (pined by the agent's devel tag)
+    if len(log_result) > 0:
+        ctx.run("git rm --ignore-unmatch {}".format(log_result))
+
+    current_branchoff = "dca-{}.{}.X".format(new_version_int[0], new_version_int[1])
+    # generate the new changelog. Specifying branch in case this is run outside the release branch that contains the tag.
+    ctx.run(
+        "reno --rel-notes-dir releasenotes-dca report \
+            --ignore-cache \
+            --branch {} \
+            --version dca-{} \
+            --no-show-source > /tmp/new_changelog-dca.rst".format(
+            current_branchoff, new_version
+        )
+    )
+
+    # reseting git
+    ctx.run("git reset --hard HEAD")
+
+    # mac's `sed` has a different syntax for the "-i" paramter
+    sed_i_arg = "-i"
+    if sys.platform == 'darwin':
+        sed_i_arg = "-i ''"
+    # remove the old header from the existing changelog
+    ctx.run("sed {0} -e '1,4d' CHANGELOG-DCA.rst".format(sed_i_arg))
+
+    if sys.platform != 'darwin':
+        # sed on darwin doesn't support `-z`. On mac, you will need to manually update the following.
+        ctx.run(
+            "sed -z {0} -e 's/dca-{1}\\n===={2}/{1}\\n{2}/' /tmp/new_changelog-dca.rst".format(
+                sed_i_arg, new_version, '=' * len(new_version)
+            )
+        )
```

Comment:
Maybe add a warning so that the release manager does not forget if they're on MacOS?

Reply:
Yeah I have this documented in our doc internally.